client/src/adapters/implementation/ChatDataAdapter.ts

```diff
@@ -1,17 +1,38 @@
 import { IChatDataAdapter } from '../interfaces/IChatDataAdapter.ts';
+import { MessageDataAdapter } from './MessageDataAdapter.ts';
+import MessageRepository from '../../api/MessageRepository.ts';
+import { Chat, ImageData, MessageData } from '../../types/types.ts';
+import ImageRepository from '../../api/ImageRepository.ts';
+import { ImageDataAdapter } from './ImageDataAdapter.ts';
 
 const ChatDataAdapter: IChatDataAdapter = {
-  getChats: (chatData) => {
-    return chatData.map((chat) => ({
-      id: chat._id,
-      chatName: '',
-      profilePicture: '',
-      lastMessage: '',
-      lastSender: '',
-      isLastMessageRead: false,
-      users: chat?.users || [],
-      messages: chat?.messages || [],
-    }));
+  getChats: async (chatData, currentUserId) => {
+    const chats: Chat[] = [];
+    for (const chat of chatData) {
```

Comment:
ja bym to zrobił bardziej funkcyjnie, czyli forEachem, ale tak też jest spoko

Reply:
forEach nie chciał współpracować z async

Reply:
```js
getChats: async (chatData, currentUserId) => {
    const chatPromises = chatData.map(async (chat) => {
    
      const [messagesResponse, imageResponse] = await Promise.all([
        MessageRepository.getAllMessagesForChat(chat._id),
        ImageRepository.getImageById(chat.chatPicture),
      ]);

      const messages = MessageDataAdapter.getMessages(
        messagesResponse.data as MessageData[],
        currentUserId,
        chat.users
      );

      const image = ImageDataAdapter.getImages(imageResponse.data as ImageData);

      return {
        id: chat._id,
        chatName: chat.name,
        chatPicture: image,
        lastMessage: '',
        lastSender: '',
        isLastMessageRead: false,
        users: chat?.users || [],
        messages: messages,
      };
    });

    const chats = await Promise.all(chatPromises);
    return chats;
  },
  ```

Reply:
teraz po dluzszym zastanowieniu imo nie jest to dobre, zeby robic w adapterze dodatkowe async calle

jak gadalismy o tym, to bardziej myslalem, ze gotowe dane przekazujemy do niego
czyli

`chatData, currentUser, chatMessages, chatPicture`

Reply:
ale jeśli to działa to ok, zostawmy już

Reply:
niezbyt moglibyśmy mieć gotowe te inne dane wcześniej, bo chaty to pierwszy krok do tego żeby wgl wgryźć się w dane